source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -314,8 +334,11 @@ namespace PlayFab
                     CallbackRequest(playFabEmitRequest, std::move(playFabEmitEventResponse));
                 }
 
-                // remove the batch from tracking map
-                this->batchesInFlight.erase(foundBatchIterator->first);
+                { // LOCK batchesInFlight mutex
+                    std::unique_lock<std::mutex> lock2(inFlightMutex);
+                    // remove the batch from tracking map
+                    this->batchesInFlight.erase(foundBatchIterator->first);
```

Comment:
`this->batchesInFlight.erase(iter->first)` is unnecessary.
just use `this->batchesInFlight.erase(iter)` (you avoid having to rehash the key and find the iterator because you already have the iterator).